core/src/main/java/io/kestra/core/schedulers/AbstractScheduler.java

```diff
@@ -412,8 +408,17 @@ private void handle() {
                                     "Polling trigger must have an interval (except the Schedule)"
                                 );
                             }
+                        } else {
+                            ZonedDateTime nextExecutionDate = null;
+                            try {
+                                nextExecutionDate = f.getPollingTrigger().nextEvaluationDate(f.getConditionContext(), Optional.of(f.getTriggerContext()));
+                            } catch (Exception e) {
+                                logError(f, e);
+                            }
+                            var trigger = f.getTriggerContext().toBuilder().nextExecutionDate(nextExecutionDate).build().checkBackfill();
+                            this.triggerState.save(trigger, scheduleContext);
```

Comment:
I don't understand why you do that here now

Reply:
If it shouldnt be evaluated, meaning the executionDate isn't validated by the condition, then we currently do nothing, then the trigger is evaluated again the next second
So if the currentDate isn't good, we do evaluate the next time it should be evaluated again

Reply:
I'm lost with all those evaluations.
I'll trust you on this one, as long as the existing tests pass.